app/common/state/immutableUtil.js

```diff
@@ -45,6 +45,22 @@ const api = {
       return defaultValue
     }
     return api.isImmutable(obj) ? obj.toJS() : obj
+  },
+
+  findNullKeyPaths (state, pathToState = []) {
+    let nullKeys = [ ]
+    if (!Immutable.Map.isMap(state) && !Immutable.List.isList(state)) {
+      return nullKeys
+    }
+    for (const key of state.keySeq()) {
+      const keyPath = [...pathToState, key]
+      if (key === null) {
+        nullKeys.push(keyPath)
+      }
+      // recursive, to find deep keys
+      nullKeys.push(...api.findNullKeyPaths(state.get(key), keyPath))
```

Comment:
recursion will be substantially more expensive than iteration to go through all the keys

Reply:
Is it worth spending the time to get this for...of...recurse loop into a flat iteration loop considering that if this code gets run the app is in an unusable state anyway?

Reply:
So... I was intrigued by the challenge, and tested what I came up with for a non-recursive function. It runs ~30% slower than the recursive one https://jsperf.com/immutable-find-null-paths-recursion-vs-iteration  Perhaps someone can improve on it.

![image](https://user-images.githubusercontent.com/741836/36627792-21e8116c-18fd-11e8-9001-959b2cda2dae.png)